planner/core/rule_column_pruning.go

```diff
@@ -106,7 +109,7 @@ func (la *LogicalAggregation) PruneColumns(parentUsedCols []*expression.Column)
 		aggrFunc.OrderByItems, cols = pruneByItems(aggrFunc.OrderByItems)
 		selfUsedCols = append(selfUsedCols, cols...)
 	}
-	if len(la.AggFuncs) == 0 {
+	if len(la.AggFuncs) == 0 || (!allFirstRow && allRemainFirstRow) {
```

Comment:
And when the `len(la.AggFuncs)` is empty, why the `allRemainFirstRow` can be true?

Reply:
These two conditions are not AND, but OR.